<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from shelf.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="shelf">
  <link name="world"/>
  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="shelf"/>
    <origin rpy="-0.007 0.002 1.5999" xyz="-0.318 -0.510 0.904"/>
  </joint>
  <link name="shelf"/>
  <link name="door">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="5.8"/>
      <inertia ixx="0.4333886585" ixy="0" ixz="0" iyy="0.39146708384" iyz="0" izz="0.4333886585"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.1475 0"/>
      <geometry>
        <box size="0.016 0.295 0.9"/>
      </geometry>
      <material name="door_material">
        <color rgba="1.0 1.0 1.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.1475 0"/>
      <geometry>
        <box size="0.016 0.295 0.9"/>
      </geometry>
    </collision>
  </link>
  <link name="door_hinge"/>
  <joint name="door_hinge_joint" type="fixed">
    <parent link="shelf"/>
    <child link="door_hinge"/>
    <dynamics damping="2.0" friction="4.0"/>
    <origin xyz="0 -0.295 0.45"/>
  </joint>
  <joint name="articulation_joint" type="revolute">
    <parent link="door_hinge"/>
    <child link="measured_joint_link"/>
    <dynamics damping="2.0" friction="4.0"/>
    <origin xyz=" -0.004 0.023 -0.150"/>
    <axis xyz="0 0 -1"/>
    <limit effort="100" lower="0" upper="1.57" velocity="10.0"/>
  </joint>
  <joint name="door_link" type="fixed">
    <parent link="measured_joint_link"/>
    <child link="door"/>
    <origin xyz="0.004 -0.023 0.150"/>
  </joint>
  <link name="handle_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.17"/>
      </geometry>
    </collision>
  </link>
  <link name="handle_support_left">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.026 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.026 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <link name="handle_support_right">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.026 0.01 0.01"/>
      </geometry>
      <material name="handle_material">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.026 0.01 0.01"/>
      </geometry>
    </collision>
  </link>
  <joint name="handle_joint" type="fixed">
    <parent link="door"/>
    <child link="handle_link"/>
    <origin rpy="0 0 0" xyz="0.034 0.27 -0.15"/>
    <!-- height handle + half the width door, 0.30 minus some offset -->
  </joint>
  <joint name="handle_support_left_joint" type="fixed">
    <parent link="handle_link"/>
    <child link="handle_support_left"/>
    <origin rpy="0 0 0" xyz="-0.013 0 0.08"/>
  </joint>
  <joint name="handle_support_right_joint" type="fixed">
    <parent link="handle_link"/>
    <child link="handle_support_right"/>
    <origin rpy="0 0 0" xyz="-0.013 0 -0.08"/>
  </joint>
  <link name="measured_joint_link"/>
  <!--     <joint name="measured_joint_handle_joint" type="fixed">
        <parent link="handle_link"/>
        <child link="measured_joint_link"/>
        <origin xyz="-0.038456463645002395 -0.24676779774627114 0.0" rpy="0 0 0"/>
    </joint> -->
</robot>

